packages/flame/lib/src/components/mixins/collidable.dart

```diff
@@ -28,6 +30,21 @@ mixin Collidable on Hitbox {
     }
     super.onRemove();
   }
+
+  @override
+  @mustCallSuper
+  void prepare(Component parent) {
```

Comment:
All of these `prepare()` methods could instead be `onLoad()`s, which run when the component is already added to the tree.

Reply:
Unfortunately not, onLoad runs only once and it will be harder for the user to catch the error as it runs later in the lifecycle and is async

Reply:
`onLoad` runs either immediately after `prepare()`, or on the next tick. How does it make any difference when the assertion is thrown?

Reply:
Hmm, maybe you are correct, add is already async so it wouldn't change much for that matter.

But still, we have the problem that `onLoad` is executed only once during the component existence, and we should run those checking everytime the component would be added to a game or new parent.

Reply:
Ah, so you mean a situation like a component is added to a game, then removed, then added once again to another game -- but this time the assertion won't be valid anymore?
This is a valid use-case, but then wouldn't the `onMount()` method be the proper place for such an assert?